worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java

```diff
@@ -30,28 +58,159 @@ public Job putJobInProgress(String jobId) {
         final Job job = jobRepository.findByJobUuid(jobId);
         Assert.notNull(job, String.format("Job %s not found", jobId));
 
-
         // validate status is SUBMITTED
         if (!SUBMITTED.equals(job.getStatus())) {
-            final String errMsg = String.format("Job %s is not in %s status. Skipping job", jobId, SUBMITTED);
+            final String errMsg = String.format("Job %s is not in %s status.", jobId, SUBMITTED);
             throw new IllegalArgumentException(errMsg);
         }
 
         job.setStatus(IN_PROGRESS);
 
-        log.info("Job [{}] is now IN_PROGRESS", job.getId());
         return jobRepository.save(job);
     }
 
     @Override
-    public void completeJob(Job job) {
+    @Transactional(propagation = Propagation.NEVER)
+    public Job processJob(final String jobId) {
+
+        final Job job = jobRepository.findByJobUuid(jobId);
+        log.info("Found job : {} - {}", job.getId(), job.getJobUuid());
+
+        final Sponsor sponsor = job.getUser().getSponsor();
+
+        final List<Contract> attestedContracts = getAggregatedAttestedContracts(sponsor);
+        log.info("Job [{}] has [{}] attested contracts", job.getJobUuid(), attestedContracts.size());
+
+
+        final Path outputDirPath = Paths.get(efsMount, job.getJobUuid());
+        final Path outputDir = fileService.createDirectory(outputDirPath);
+
+        for (Contract contract : attestedContracts) {
+            log.info("Job [{}] - contract [{}] ", job.getJobUuid(), contract.getContractNumber());
+
+            var jobOutputs = processContract(outputDir, contract);
+
+            jobOutputs.forEach(jobOutput -> job.addJobOutput(jobOutput));
+            jobOutputRepository.saveAll(jobOutputs);
+        }
+
+        completeJob(job);
+        return job;
+    }
+
+
+    private List<Contract> getAggregatedAttestedContracts(Sponsor sponsor) {
+         if (sponsor.getParent() == null) {
+             // implies this sponsor is a parent sponsor. Parent sponsors do not have contracts.
+             // Hence, find all the children and process their contracts instead
+             log.info("Sponsor {} is a parent sponsor. Processing children sponsors", sponsor.getOrgName());
+
+             return getContractsOfChildrenSponsor(sponsor);
+         } else {
+            return sponsor.getAttestedContracts();
+         }
+    }
+
+
+    private List<Contract> getContractsOfChildrenSponsor(Sponsor sponsor) {
+        final List<Sponsor> childrenSponsors = sponsorRepository.findByParent(sponsor);
+        return childrenSponsors.stream()
+                .map(s -> s.getAttestedContracts())
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+    }
```

Comment:
My goal was to have this logic encapsulated in the domain class itself. Asked Monica to create a tech debt ticket to address this.

Reply:
hmm. I needed to make another query using a repository.
```sponsorRepository.findByParent(sponsor);```
which is why I ended up putting it in the service layer.

Reply:
Right. We might need to introduce `Sponsor.children` JPA-managed collection.

Reply:
sounds good